src/crh/constraints.rs

```diff
@@ -1,12 +1,12 @@
 use ark_ff::Field;
 use core::fmt::Debug;
 
-use crate::crh::FixedLengthCRH;
+use crate::crh::{TwoToOneCRH, CRH};
 use ark_relations::r1cs::SynthesisError;
 
 use ark_r1cs_std::prelude::*;
 
-pub trait FixedLengthCRHGadget<H: FixedLengthCRH, ConstraintF: Field>: Sized {
+pub trait CRHGadget<H: CRH, ConstraintF: Field>: Sized {
```

Comment:
Maybe we should add some documentation indicating that this is still fixed length? (If not, then we should still have a `FixedLengthCRHGadget`

Reply:
May be we need to decide which of the following set of names is good. 
- Choice 1: `FixedLengthCRH`, `TwoToOneFixedLengthCRH`
- Choice 2: `CRH` (name might collide with implementation name if we choose to use something like `pedersen::CRH`), `FixedLengthCRH` with additional documentation showing that this is still fixed length. 
- Choice 3: something else...
Which one is better? cc: @ValarDragon @weikengchen

Reply:
resolved